nixos/tests/appliance-gpt-image.nix

```diff
@@ -0,0 +1,114 @@
+# Tests building and running a GUID Partition Table (GPT) appliance image.
+# "Appliance" here means that the image does not contain the normal NixOS
+# infrastructure of a system profile and cannot be re-built via
+# `nixos-rebuild`.
+
+{ lib, ... }:
+
+let
+  rootParitionlabel = "root";
+
+  bootLoaderConfigPath = "/loader/entries/nixos.conf";
+  kernelPath = "/EFI/nixos/kernel.efi";
+  initrdPath = "/EFI/nixos/initrd.efi";
+in
+{
+  name = "appliance-gpt-image";
+
+  meta.maintainers = with lib.maintainers; [ nikstur ];
+
+  nodes.machine = { config, lib, pkgs, ... }: {
+
+    imports = [ ../modules/image/gpt.nix ];
+
+    virtualisation.directBoot.enable = false;
+    virtualisation.mountHostNixStore = false;
+    virtualisation.useEFIBoot = true;
+
+    # Disable boot loaders because we install one "manually".
+    boot.loader.grub.enable = false;
```

Comment:
In the future, a GPT image appliance will be able to provide bootloaders via #244907 by having a `appliance-loader` or you can add compatibility with that. Therefore, we will still be able to validate that a bootloader is contained inside the appliance.

Reply:
Cool, let's remove this once https://github.com/NixOS/nixpkgs/pull/244907 lands.

Reply:
Just add a `# TODO(raitobezarius): revisit this when #244907 lands`